src/formats/yarle/utils/folder-utils.ts

```diff
@@ -119,7 +120,20 @@ export const clearResourceDir = (note: any): void => {
 	resourceDirClears.set(resPath, clears + 1);
 };
 
-export const setPaths = (enexSource: PickedFile): void => {
+export const handleNotebookstacks = (enex: PickedFile, options: YarleOptions): void => {
+	const notebookFolderNames = enex.basename.split('@@@');
+	let notebookName = notebookFolderNames.pop();
+	if (!notebookName){
+		notebookName = enex.basename;
+	}
+	fs.mkdirSync(path.join(options.outputDir, ...notebookFolderNames), { recursive: true });
+	enex.fullpath = enex.fullpath.replace(enex.basename, notebookName || enex.basename);
+	enex.name = enex.name.replace(enex.basename, notebookName || enex.basename);
+	options.outputDir = [options.outputDir, ...notebookFolderNames].join(options.pathSeparator);
```

Comment:
Will this behave as expected if there are multiple files being imported?

It looks like if we modify the outputDir here it will affect subsequent files being imported.

Reply:
Nope, because I save the original outputDir  at line 187 in yarle.ts and restore after the conversion (line 201).

Reply:
I believe that's just saving a reference to the object, but that's the same object that will be modified here.

Reply:
Ah, sorry I misread and thought you were storing the full options object.
Thank you for clarifying for me!

Reply:
@lishid ok, let me try to improve it.